src/vos/vos_obj.c

```diff
@@ -947,6 +958,9 @@ key_iter_probe(struct vos_obj_iter *oiter, daos_anchor_t *anchor)
 {
 	int	rc;
 
+	/** If we are reprobing, assume we've skipped something */
```

Comment:
I think reprobing will always happen since iteration has to yield from time to time, that means the it_skipped will always be set and the 'need_agg" flag will never be cleared?

Reply:
Maybe we don't need this.  If we instead track whether we set the AGG_FLAG, we can avoid clearing NEEDED flag if it's a new write